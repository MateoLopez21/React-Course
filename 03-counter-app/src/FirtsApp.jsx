import PropTypes from 'prop-types';

const newSaludo = (name) => {
    return `Hola, ${name}`
}

// const newMessage = 'Mateo LÃ³pez';
export const FirtsApp = ( { name } ) => {


    return (
        <>
            <h1> { newSaludo(name) } </h1>
            <p>Lorem ipsum dolor sit amet consectetur, adipisicing elit. Expedita, tempora? Sapiente nisi iure perferendis est esse quam aliquam odio aliquid soluta deleniti sunt, dolorum ipsam nihil sint vero omnis placeat.</p>
        </>
    )
}


FirtsApp.propTypes = {
    name: PropTypes.string.isRequired
}

FirtsApp.defaultProps = {
    name: 'No hay nombre'
}