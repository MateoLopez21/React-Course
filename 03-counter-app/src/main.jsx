import React from 'react'
import ReactDOM from 'react-dom/client'
// import { HelloWorldApp } from './HelloWorldApp'
// import { FirtsApp } from './FirtsApp'
import './index.css'
import { CounterApp } from './CounterApp'

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        {/* <HelloWorldApp /> */}
        {/* <FirtsApp name='Mateo LÃ³pez' /> */}
        <CounterApp value={ 0 } />
    </React.StrictMode>
)


