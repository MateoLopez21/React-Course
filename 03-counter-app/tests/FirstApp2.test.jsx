import {render, screen} from '@testing-library/react'
import { FirtsApp } from '../src/FirtsApp'

describe('Pruebas en <FirstApp />', () => { 

    const name = 'Mateo López'

    test('Debe de hacer match con el snapshot', () => { 

        const { container } = render(<FirtsApp name={ name } />);

        expect( container ).toMatchSnapshot();

     })
    test('Debe de mostrar el nombre "Mateo López" ', () => { 

      render(<FirtsApp name={ name } />);

      expect( screen.getByText(name)).toBeTruthy();

     })
    test('Debe de mostrar el nombre en un h1', () => { 

      render(<FirtsApp name={ name } />);

      expect( screen.getByRole('heading', { level: 1}).innerHTML).toContain(name);

     })
    test('Debe de mostrar el nombre enviado por props', () => { 

      render(<FirtsApp name={ name } />);

      expect(screen.getAllByText(name)).toBeTruthy();

     })
 })