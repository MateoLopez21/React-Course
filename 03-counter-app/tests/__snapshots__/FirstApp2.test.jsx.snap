// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-name"
  >
     
    Mateo LÃ³pez
     
  </h1>
  <p>
    Lorem ipsum dolor sit amet consectetur, adipisicing elit. Expedita, tempora? Sapiente nisi iure perferendis est esse quam aliquam odio aliquid soluta deleniti sunt, dolorum ipsam nihil sint vero omnis placeat.
  </p>
</div>
`;
